src/command/repo/get.c

```diff
@@ -46,13 +49,81 @@ storageGetProcess(IoWrite *destination)
 
             if (repoCipherType != cipherTypeNone)
             {
+                /*******************************************************************************************************************
+                Repository encryption design:
+                REPO / (repoCipherPass)
+                     / archive   / (repoCipherPass)
+                     / archive   / stanza / (archiveCipherPass)
+                     / backup    / (repoCipherPass)
+                     / backup    / stanza / (backupCipherPass)
+                     / backup    / stanza / set / (manifestCipherPass)
+                     / backup    / stanza / backup.history / (manifestCipherPass)
+                *******************************************************************************************************************/
+
                 // Check for a passphrase parameter
                 const String *cipherPass = cfgOptionStrNull(cfgOptCipherPass);
 
-                // If not passed as a parameter use the repo passphrase
+                // If not passed as a parameter, find the passphrase
                 if (cipherPass == NULL)
+                {
                     cipherPass = cfgOptionStr(cfgOptRepoCipherPass);
 
+                    const String *relativeFile = strDup(file);
+                    if (strBeginsWith(relativeFile, FSLASH_STR))
+                        relativeFile = strSub(file, strSize(cfgOptionStr(cfgOptRepoPath)) + 1);
+
+                    // Encrypted files could be stored directly at the root of the repo if needed
+                    // We should then at least be able to determine the archive or backup directory
+                    StringList *filePathSplitLst = strLstNewSplit(relativeFile, FSLASH_STR);
+                    if (strLstSize(filePathSplitLst) > 1)
+                    {
+                        const String *stanza = strLstGet(filePathSplitLst, 1);
+
+                        // If stanza option is specified, it must match the given file path
+                        if (cfgOptionStrNull(cfgOptStanza) != NULL && ! strEq(stanza, cfgOptionStr(cfgOptStanza)))
+                            THROW_FMT(AssertError, "stanza name '%s' given in option doesn't match the given path", 
```

Comment:
If more than one line follows an if then add braces to make it clearer: https://github.com/pgbackrest/pgbackrest/blob/master/CODING.md#braces.

Also there should be a LF after `THROW_FMT(`.

Reply:
fixed, thanks